R/fit.R

```diff
@@ -66,6 +66,51 @@ CmdStanFit <- R6::R6Class(
   )
 )
 
+#' Extract gradients of the diagnostic
+#'
+#' @name fit-method-gradients
+#' @aliases gradients
+#' @description Return the data frame containing the gradients
+#'   for all parameters in the 
+#'
+#' @return A list of lists. See **Examples**.
+#'
+#' @seealso [`CmdStanDiagnose`]
+#'
+#' @examples
+#' \dontrun{
+#' fit <- cmdstanr_example("logistic", method = "diagnose")
+#'
+#' # retrieve the gradients
+#' fit$gradients()
+#' }
+#'
+CmdStanDiagnose <- R6::R6Class(
+  classname = "CmdStanDiagnose",
```

Comment:
This looks like doc for the `gradients` method but I think we also need doc for CmdStanDiagnose object itself like we have for all the other CmdStan* objects (i.e., a page where `?CmdStanDiagnose` would take you where it lists available methods, etc.). I think you can pretty much copy what we did for the other methods. Does that make sense?

Reply:
If you want me to add this let me know, I might have some time tomorrow to do it.

Reply:
I think I get what you mean, maybe :) If you have time that would be fantastic.

Reply:
Done!